### DISCUSSION TITLE
Need help resolving plotting error when migrating from Shapely 1.8 to 2.0.

### DISCUSSION THREAD

**Discussion by olboyd-usgs** [shapely/shapely#1710] [Q&A] [answered]
Hello,
The following code no longer works with Shapely 2.0, and PolygonPatch (from descartes) generates the error "_IndexError: too many indices for array: array is 0-dimensional, but 2 were indexed_". Note that I had to reference mp.geoms in the new version of Shapely, but I wonder if there's something else that needs to be done to account for changes in the new version.

    import fiona
    from matplotlib.collections import PatchCollection
    from descartes import PolygonPatch
    from shapely.geometry import Polygon, MultiPolygon, shape
    import numpy as np
   
    startTime = datetime.now()
    patches = []
    mp = MultiPolygon(
        [shape(pol['geometry']) for pol in fiona.open(CartDir + 'States/st99_d00.shp')
        if pol['properties']['AREA'] > 0.01])
    for idx, p in enumerate(mp.geoms):
        patches.append(PolygonPatch(p, fill=False, ec='#555555', zorder=1))

Thanks for any help whomever is able to provide,
Oliver

**Comment by jorisvandenbossche** [comment, marked as answer]
The main issue is that `descartes` is not compatible with shapely 2.0, but `descartes` is also basically a dead/archived project, and so probably won't receive any updates anymore (https://pypi.org/project/descartes/ links to bitbucket that doesn't exist anymore). Although updating it to be compatible with shapely 2.0 should be quite easy ..

Now, the good part is that it is easy to replace `descartes` in your code, since it is only a small package. For example, GeoPandas did this (https://github.com/geopandas/geopandas/pull/1677), and you can see that the actual `PolygonPatch` is only a few lines of code.

In addition, starting with shapely 2.0, we also added the plotting capabilities of descartes directly to shapely itself. So if you can require shapely >= 2.0, you can also use that: 

```
from shapely.plotting import plot_polygon

....

for idx, p in enumerate(mp.geoms):
    plot_polygon(p, fill=False, ec='#555555', zorder=1)

# or for constant styling arguments, plot multipolygon directly
plot_polygon(mp, fill=False, ec='#555555', zorder=1)
```

Those function directly plot on a matplotlib figure/axes. If you would prefer to handle the patches manually, there is currently a PR to expose this helper function as well: https://github.com/shapely/shapely/pull/1704 (and this is basically exactly the PolygonPatch from descartes)

**Reply by olboyd-usgs** [reply]
Thank you for the suggestions. Since I am using both matplotlib and plotly, I prefer a function that returns the patches, which I've done successfully using using the helper function patch_from_polygon that you referenced in #1704.
